pkg/client/fake/client.go

```diff
@@ -598,22 +598,23 @@ func (c *fakeClient) List(ctx context.Context, obj client.ObjectList, opts ...cl
 	if err != nil {
 		return err
 	}
-	zero(obj)
-	if err := json.Unmarshal(j, obj); err != nil {
+	objCopy := obj.DeepCopyObject().(client.ObjectList)
+	zero(objCopy)
```

Comment:
Are we missing something by not zero'ing obj anymore?

Reply:
Not sure why I did it this way, changed it to just keep zeroing `obj` instead, ptal